src/devices/bus/nes_ctrl/turbofile.cpp

```diff
@@ -0,0 +1,130 @@
+// license:BSD-3-Clause
+// copyright-holders:kmg
+/**********************************************************************
+
+    Nintendo Family Computer ASCII Turbofile SRAM external storage device
+
+    The original 8K Turbofile, emulated here, has been seen with serials
+    AS-TF02 through AS-TF04, differences unknown. We do not support the
+    Turbo File II (yet?), but it appears to simply be a 32K version with
+    a manual switch to select between four 8K banks. Both devices also
+    sport a write protect switch.
+
+**********************************************************************/
+
+#include "emu.h"
+#include "turbofile.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(NES_TURBOFILE, nes_turbofile_device, "nes_turbofile", "ASCII Turbofile Backup RAM")
+
+
+INPUT_CHANGED_MEMBER( nes_turbofile_device::lock_changed )
+{
+	m_locked ^= 1;
+	popmessage("Turbofile Protect Switch %s\n", m_locked ? "ON" : "OFF");
+}
```

Comment:
Since you’ve got a toggle input field, you should just use the new value, not arbitrarily flip the value, or it will get out of sync.

Reply:
Fixed.